aws-rds-cfn-common/src/main/java/software/amazon/rds/common/handler/Commons.java

```diff
@@ -77,10 +77,9 @@ public static <M, C> ProgressEvent<M, C> execOnce(
             final VoidBiFunction<C, Boolean> conditionSetter
     ) {
         if (!conditionGetter.apply(progress.getCallbackContext())) {
-            return func.enact().then(p -> {
-                conditionSetter.apply(p.getCallbackContext(), true);
-                return p;
-            });
+            ProgressEvent<M, C> progressEvent = func.enact();
+            conditionSetter.apply(progressEvent.getCallbackContext(), true);
```

Comment:
In case then the `func` is executed more than once (e.g. stabilisation) this will cause this code to exit prematurely. I think it is safer to throw if progress delay is not zero.

Reply:
Unfortunately, a stabilization scenario may fail and throw exception if delay was not 0 as well. I updated it.